raml-parser-2/src/main/java/org/raml/v2/internal/impl/commons/model/type/TypeDeclaration.java

```diff
@@ -90,6 +89,42 @@ public String type()
         return getTypeExpression(getTypeNode());
     }
 
+    public List<TypeDeclaration> parentTypes() {
+
+        List<TypeDeclaration> nodes =  new ArrayList<>();
+        Set<String> parentNames = new HashSet<>();
+        Node parents =  resolvedType.getTypeDeclarationNode().get("type");
+        if ( parents == null) {
+
+           return nodes;
+        }
+
+        List<Node> parentTypeNodes = parents.getChildren();
+        for (Node parentTypeNode : parentTypeNodes) {
+            NamedTypeExpressionNode parent = (NamedTypeExpressionNode) parentTypeNode;
+            parentNames.add(parent.getValue());
+        }
+
+        Node nn = node.getRootNode().get("types");
+        if ( nn == null ) {
+            return nodes;
+        }
+
+        TypeDeclarationModelFactory factory = new TypeDeclarationModelFactory();
```

Comment:
That should do it for you all type of  reference should be resolved. 
Just one comment on your approach reference can be not only to local types but it can also be to libraries, that is why we have a specific phase and logic to resolve this.
Hope this clarifies let me know if not
Regards